lib/commons/aria/implicit-role.js

```diff
@@ -9,86 +9,37 @@ import lookupTable from './lookup-table';
  * @param {HTMLElement} node The node to test
  * @return {Mixed} Either the role or `null` if there is none
  */
-// TODO: axe 4.0 - rename to `getImplicitRole` so we can use
-// a variable named `implicitRole` when assigning the output of
-// the function
 function implicitRole(node) {
-	'use strict';
-
-	/*
-	 * Filter function to reduce a list of roles to a valid list of roles for a nodetype
-	 */
-	const isValidImplicitRole = function(set, role) {
-		const validForNodeType = function(implicitNodeTypeSelector) {
-			// TODO: es-module-utils.matchesSelector
-			return axe.utils.matchesSelector(node, implicitNodeTypeSelector);
-		};
-
-		if (role.implicit && role.implicit.some(validForNodeType)) {
-			set.push(role.name);
-		}
-
-		return set;
-	};
-
-	/*
-	 * Score a set of roles and aria-attributes by its optimal score
-	 * E.g. [{score: 2, name: button}, {score: 1, name: main}]
-	 */
-	const sortRolesByOptimalAriaContext = function(roles, ariaAttributes) {
-		const getScore = function(role) {
-			const allowedAriaAttributes = allowedAttr(role);
-			return allowedAriaAttributes.reduce(function(score, attribute) {
-				return score + (ariaAttributes.indexOf(attribute) > -1 ? 1 : 0);
-			}, 0);
-		};
-
-		const scored = roles.map(function(role) {
-			return { score: getScore(role), name: role };
-		});
-
-		const sorted = scored.sort(function(scoredRoleA, scoredRoleB) {
-			return scoredRoleB.score - scoredRoleA.score;
-		});
-
-		return sorted.map(function(sortedRole) {
-			return sortedRole.name;
-		});
-	};
+	const vNode = getNodeFromTree(node);
+
+	// this error is only thrown if the virtual tree is not a
+	// complete tree, which only happens in linting and if a
+	// user used `getFlattenedTree` manually on a subset of the
+	// DOM tree
+	if (!vNode) {
+		throw new ReferenceError(
+			'Cannot get implicit role of a node outside the current scope.'
+		);
+	}
 
-	/*
-	 * Create a list of { name / implicit } role mappings to filter on
-	 */
-	const roles = Object.keys(lookupTable.role).map(function(role) {
-		const lookup = lookupTable.role[role];
-		return { name: role, implicit: lookup && lookup.implicit };
-	});
+	// until we have proper implicit role lookups for svgs we will
+	// avoid giving them one
+	if (node.namespaceURI === 'http://www.w3.org/2000/svg') {
+		return null;
+	}
 
-	/* Build a list of valid implicit roles for this node */
-	const availableImplicitRoles = roles.reduce(isValidImplicitRole, []);
+	const nodeName = vNode.props.nodeName;
+	const role = lookupTable.implicitRole[nodeName];
```

Comment:
I still think this is going to be a problem with SVG later. I'd think we need to rename this prop:

```suggestion
	const role = lookupTable.implicitHtmlRole[nodeName];
```